src/metabase/util/schema.clj

```diff
@@ -291,56 +389,132 @@
+  (with-error-message-malli
+    [:and :string [:fn u/email?]]
+    (deferred-tru "value must be a valid email address.")))
+
 (def ValidPassword
   "Schema for a valid password of sufficient complexity which is not found on a common password list."
   (with-api-error-message (s/constrained s/Str u.password/is-valid?)
     (deferred-tru "password is too common.")))
 
+(defschema ValidPasswordMalli
+  "Schema for a valid password of sufficient complexity which is not found on a common password list."
+  (with-error-message-malli
+    [:and
+     :string
+     [:fn u.password/is-valid?]]
+    (fn [_ _] (deferred-tru "password is too common."))))
+
 (def IntString
   "Schema for a string that can be parsed as an integer.
    Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
   (with-api-error-message (s/constrained s/Str #(u/ignore-exceptions (Integer/parseInt %)))
     (deferred-tru "value must be a valid integer.")))
 
+(defschema IntStringMalli
+  "Schema for a string that can be parsed as an integer.
+  Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
+  (with-error-message-malli
+    [:and
+     :string
+     [:fn #(u/ignore-exceptions (Integer/parseInt %))]]
+    (deferred-tru "value must be a valid integer.")))
+
 (def IntStringGreaterThanZero
   "Schema for a string that can be parsed as an integer, and is greater than zero.
    Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
   (with-api-error-message (s/constrained s/Str #(u/ignore-exceptions (< 0 (Integer/parseInt %))))
     (deferred-tru "value must be a valid integer greater than zero.")))
 
+(defschema IntStringGreaterThanZeroMalli
+  "Schema for a string that can be parsed as an integer, and is greater than zero.
+  Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
+  (with-error-message-malli
+    [:and
+     :string
+     [:fn #(u/ignore-exceptions (< 0 (Integer/parseInt %)))]]
+    (deferred-tru "value must be a valid integer greater than zero.")))
+
 (def IntStringGreaterThanOrEqualToZero
   "Schema for a string that can be parsed as an integer, and is greater than or equal to zero.
    Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
   (with-api-error-message (s/constrained s/Str #(u/ignore-exceptions (<= 0 (Integer/parseInt %))))
     (deferred-tru "value must be a valid integer greater than or equal to zero.")))
 
+(defschema IntStringGreaterThanOrEqualToZeroMalli
+  "Schema for a string that can be parsed as an integer, and is greater than or equal to zero.
+  Something that adheres to this schema is guaranteed to to work with `Integer/parseInt`."
+  (with-error-message-malli
+    [:and
+     :string
+     [:fn #(u/ignore-exceptions (<= 0 (Integer/parseInt %)))]]
+    (deferred-tru "value must be a valid integer greater than or equal to zero.")))
```

Comment:
Consider this as an example. This can be done with vanilla malli, so I don't see the purpose of `with-error-message-malli`.

```suggestion
[:and {:error/fn (constantly (deferred-tru "value must be a valid integer greater than or equal to zero."))}
   :string
   [:fn #(u/ignore-exceptions (<= 0 (Integer/parseInt %)))]])
```